xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less

```diff
@@ -175,7 +175,8 @@ body {
       & .xtree,
       & .xwikitabbar,
       & .buttonwrapper,
-      & #user-menu-col {
+      & #user-menu-col,
+      & .jstree-xwiki {
```

Comment:
We have that class iirc, but I think this is the never-ending debate of semantic classes vs. classes that encode styles. As a compromise, maybe we could move these styles to where we define the other styles for jstree?

Reply:
Yup, we have a class: https://github.com/xwiki/xwiki-platform/blob/dc1103408db92c0fc588448fcbf0c0b7db22dcec/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less#L205

I'm trying to put in here only the classes that are generic enough, and only use this styling class when I find an outlier. 

For this example, we don't want any jstree to have underlined links, so I figured it'd make sense to add it to the list here (so that we don't have to come back to a very similar issue in a month when we find another exception).

> I'm afraid we are going to end up with an infinite list of selectors that is going to be hard to maintain.

IMO we should only put selectors needed for the xwiki standard flavor here. AFAIK there should be a couple more items in there at most. Most of the leftover underlining issues I saw are specific and should be solved with the styling class. Maybe I should add some comments on each line to make sure it's easy to understand and maintain those lists?

> As a compromise, maybe we could move these styles to where we define the other styles for jstree?

I'm strongly against this. I'd rather keep all the CSS related to this style preference in one place. Having to go through (and remember) multiple files to understand how each element interacts with this preference would make things difficult to maintain.

Reply:
I went too fast and didn't see the `.xtree` line a bit above the changes. Seems like this item was already in the list, but the exact class used wasn't generic enough to cover all the trees.

Reply:
Done in  https://github.com/xwiki/xwiki-platform/pull/3158/commits/c067e94d8610de1ffc213a802a8c60b95056d261